client/components/import/import.js

```diff
@@ -11,48 +11,105 @@ const ImportPopup = BlazeComponent.extendComponent({
     return 'importPopup';
   },
 
-  events() {
-    return [{
-      'submit': (evt) => {
-        evt.preventDefault();
-        const dataJson = $(evt.currentTarget).find('.js-import-json').val();
-        let dataObject;
-        try {
-          dataObject = JSON.parse(dataJson);
-        } catch (e) {
-          this.setError('error-json-malformed');
-          return;
-        }
-        Meteor.call(this.getMethodName(), dataObject, this.getAdditionalData(),
-          (error, response) => {
-            if (error) {
-              this.setError(error.error);
-            } else {
-              Filter.addException(response);
-              this.onFinish(response);
-            }
-          }
-        );
-      },
-    }];
+  jsonText() {
+    return Session.get('import.text');
+  },
+
+  membersMapping() {
+    return Session.get('import.membersToMap');
   },
 
   onCreated() {
     this.error = new ReactiveVar('');
+    this.dataToImport = '';
+  },
+
+  onFinish() {
+    Popup.close();
+  },
+
+  onShowMapping(evt) {
+    this._storeText(evt);
+    Popup.open('mapMembers')(evt);
+  },
+
+  _storeText() {
+    const dataJson = this.$('.js-import-json').val();
+    Session.set('import.text', dataJson);
+    return dataJson;
+  },
+
+  onSubmit(evt){
+    evt.preventDefault();
+    const dataJson = this._storeText(evt);
+    let dataObject;
+    try {
+      dataObject = JSON.parse(dataJson);
+      this.setError('');
+    } catch (e) {
+      this.setError('error-json-malformed');
+      return;
+    }
+    // if there are members listed in the import and we have no mapping for them...
+    if(dataObject.members.length > 0 && !this.membersMapping()) {
+      // we will work on the list itself (an ordered array of POJO)
```

Comment:
[POJO](https://fr.wikipedia.org/wiki/Plain_Old_Java_Object)? I’m not familiar with this acronym, I would rather use “JavaScript Object” (or “Dictionary” as in Python, but the “JavaScript Object” is clearer).